XCUITest/SearchSuggestionsTest.swift

```diff
@@ -0,0 +1,170 @@
+
+class SearchSuggestionsPromptTest: BaseTestCase {
+    
+    override func setUp() {
+        super.setUp()
+        dismissFirstRunUI()
+    }
+    
+    override func tearDown() {
+        app.terminate()
+        super.tearDown()
+    }
+    
+    func checkToggle(isOn: Bool) {
+        let targetValue = isOn ? "1" : "0"
+        XCTAssertEqual(app.tables.switches["BlockerToggle.enableSearchSuggestions"].value as! String, targetValue)
+    }
+    
+    func checkSuggestions() {
+        // Check search cells are displayed correctly
+        let firstSuggestion = app.buttons.matching(identifier: "OverlayView.searchButton").element(boundBy: 0)
+        let secondSuggestion = app.buttons.matching(identifier: "OverlayView.searchButton").element(boundBy: 1)
+        let thirdSuggestion = app.buttons.matching(identifier: "OverlayView.searchButton").element(boundBy: 2)
+        let fourthSuggestion = app.buttons.matching(identifier: "OverlayView.searchButton").element(boundBy: 3)
+        
+        waitforExistence(element: firstSuggestion)
+        waitforExistence(element: secondSuggestion)
+        waitforExistence(element: thirdSuggestion)
+        waitforExistence(element: fourthSuggestion)
+        
+        XCTAssertEqual("g", firstSuggestion.label)
+        XCTAssertEqual("gmail", secondSuggestion.label)
+        XCTAssertEqual("google", thirdSuggestion.label)
+        XCTAssertEqual("google maps", fourthSuggestion.label)
+        
+        // Check tapping on suggestion leads to correct page
+        firstSuggestion.tap()
+        waitForValueContains(element: app.textFields["URLBar.urlText"], value: "g")
+    }
+    
+    func typeInURLBar(text: String) {
+        app.textFields["Search or enter address"].tap()
+        app.textFields["Search or enter address"].typeText(text)
+    }
+    
+    func testEnableThroughPrompt() {
+        // Check search suggestions toggle is initially OFF
+        waitforHittable(element: app.buttons["Settings"])
+        app.buttons["Settings"].tap()
+        checkToggle(isOn: false)
+        
+        // Activate prompt by typing in URL bar
+        app.buttons["SettingsViewController.doneButton"].tap()
+        typeInURLBar(text: "g")
+        
+        // Prompt should display
+        waitforExistence(element: app.otherElements["SearchSuggestionsPromptView"])
+        
+        // Press enable
+        app.buttons["SearchSuggestionsPromptView.enableButton"].tap()
+
+        // Ensure prompt disappears
+        waitforNoExistence(element: app.otherElements["SearchSuggestionsPromptView"])
+        
+        // Ensure search suggestions are shown
+        checkSuggestions()
+
+        // Check search suggestions toggle is ON
+        waitforHittable(element: app.buttons["HomeView.settingsButton"])
+        app.buttons["HomeView.settingsButton"].tap()
+        checkToggle(isOn: true)
+    }
+    
+    func testDisableThroughPrompt() {
+        // Check search suggestions toggle is initially OFF
```

Comment:
It seems we repeat this in every function. It should probably be broken out into its own function and then just called (like how you've done with typeInURLBar)